PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/LaundryRecyclerAdapter.java

```diff
@@ -0,0 +1,308 @@
+                        snackTextView.setTextColor(activity.getResources().getColor(R.color.white));
+                        snackbar.show();
+                    } else {
+                        final PendingIntent alarmIntent = PendingIntent.getBroadcast(activity, laundryRoom.name.hashCode() + machine.number,
+                                intent, PendingIntent.FLAG_NO_CREATE);
+                        if (alarmIntent != null) {
+                            alarmManager.cancel(alarmIntent);
+                            alarmIntent.cancel();
+                        }
+                        StringBuilder stringBuilder = new StringBuilder();
+                        stringBuilder.append(activity.getString(R.string.laundry_notification_snackbar_off))
+                                .append(" ").append(machine.machine_type).append(" ")
+                                .append(machine.number);
+                        if (buttonView.getContext() == null) {
+                            return;
+                        }
+                        Snackbar snackbar = Snackbar.make(buttonView,stringBuilder, Snackbar.LENGTH_SHORT);
+                        View subView = snackbar.getView();
+                        TextView snackTextView = (TextView) subView.findViewById(android.support.design.R.id.snackbar_text);
+                        snackTextView.setTextColor(activity.getResources().getColor(R.color.white));
+                        snackbar.show();
+                    }
+                }
+            });
+        }
+
+        public void bindMachineStatusHolder(LaundryMachine machine, LaundryRoom laundryRoom) {
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append(machine.machine_type).append(" ").append(machine.number);
+            title.setText(stringBuilder);
+            if (machine.available) {
+                description.setText(R.string.laundry_available);
+                description.setTextColor(activity.getResources().getColor(R.color.avail_color_green));
+                notification.setVisibility(View.GONE);
+            } else {
+                stringBuilder = new StringBuilder();
+                stringBuilder.append("Busy – ").append(machine.getTimeLeft(activity));
+                description.setText(stringBuilder);
+                if (machine.getTimeInMilli() > 0) {
+                    notification.setVisibility(View.VISIBLE);
+                } else {
+                    notification.setVisibility(View.GONE);
+                }
+                setSwitchState(machine, notification, laundryRoom, activity);
+                description.setTextColor(activity.getResources().getColor(R.color.avail_color_red));
+            }
+        }
+    }
+
+    private List<LaundryMachine> machines;
+    private MainActivity activity;
+    private boolean wash;
+    private LaundryRoom laundryRoom;
+    private int[] traffic;
+
```

Comment:
Move this/ the TimeXAxisValueFormatter into a new class... these private variables shouldn't be all over the place